internal/util/parse/parse_test.go

```diff
@@ -182,3 +183,33 @@ func Test_pkgURLFromGHURL(t *testing.T) {
 		})
 	}
 }
+
+func Test_parseURLL(t *testing.T) {
+	type expected struct {
+		repo    string
+		dir     string
+		version string
+	}
+	tests := map[string]struct {
+		ghURL string
+		expected
+	}{
+		"simple": {
+			ghURL:    "https://my.git-server.com/repository.git/directory@main",
+			expected: expected{repo: "https://my.git-server.com/repository", dir: "/directory", version: "main"},
```

Comment:
we should actually keep the .git part of the repository, shouldn't we?

with GitHub it happens to work that removing .git suffix points at the same repo but there is no such guarantee in the general case and the `.git` is part of the input URL, seems incorrect to remove it.

I'd expect output:
repo: https://my.git-server.com/repository.git
dir: /directory
version: main

Reply:
I'm not that familiar with this code, but I find that keeping `.git` here causes some pkg update tests to fail. I suspect that kpt is appending the .git suffix elsewhere (or doing some other magic to make it work). 

I'll look into a bit more to make sure, but I think the test for porch that uses the URL `http://git-server.test-git-namespace.svc.cluster.local:8080.git/@main` should catch any problems with nongithub URLs.

Reply:
The URL used in Porch is an unfortunate hack I had to put in place due to the parsing bug so we can update it.

the URL I'd like to be using is:

`http://git-server.test-git-namespace.svc.cluster.local:8080/@main`

There is a separate problem where kpt will actually try to clone repository as part of parsing the address to glean where the separation between repository address and directory path within the repository is.

Reply:
Sorry for the delay, just getting back to this now.

I see the porch test URLs have been updated in main, I'm assuming they're now the URLs that you want so there's no need to update them in this PR anymore.

Reply:
correct.

Reply:
> we should actually keep the .git part of the repository, shouldn't we?
> 
> with GitHub it happens to work that removing .git suffix points at the same repo but there is no such guarantee in the general case and the .git is part of the input URL, seems incorrect to remove it.
> 
> I'd expect output:
> repo: https://my.git-server.com/repository.git
> dir: /directory
> version: main

I checked this code with a local filepath `kpt pkg get /path/to/repo.git` and it was successful. Is that enough to give us confidence that this will work with nongithub URLs?

Reply:
This is great, thank you!